src/LuaTransformer.ts

```diff
@@ -4646,6 +4648,60 @@ export class LuaTransformer {
         return this.createSelfIdentifier(thisKeyword);
     }
 
+    public transformTaggedTemplateExpression(expression: ts.TaggedTemplateExpression): ExpressionVisitResult {
+        const strings: string[] = [];
+        const rawStrings: string[] = [];
+        const expressions: ts.Expression[] = [];
+
+        if (ts.isTemplateExpression(expression.template)) {
+            // Expressions are in the string.
+            strings.push(expression.template.head.text);
+            rawStrings.push(tsHelper.getRawLiteral(expression.template.head));
+            strings.push(...expression.template.templateSpans.map(span => span.literal.text));
+            rawStrings.push(...expression.template.templateSpans.map(span => tsHelper.getRawLiteral(span.literal)));
+            expressions.push(...expression.template.templateSpans.map(span => span.expression));
+        } else {
+            // No expressions are in the string.
+            strings.push(expression.template.text);
+            rawStrings.push(tsHelper.getRawLiteral(expression.template));
+        }
+
+        // Construct string arrays for parameters.
+        const stringArray = ts.createArrayLiteral(strings.map(partialString => ts.createStringLiteral(partialString)));
+
+        // Evaluate if there is a self parameter to be used.
+        const signature = this.checker.getResolvedSignature(expression);
+        const signatureDeclaration = signature && signature.getDeclaration();
+        const useSelfParameter =
+            signatureDeclaration &&
+            tsHelper.getDeclarationContextType(signatureDeclaration, this.checker) !== ContextType.Void;
+        const context = useSelfParameter ? (this.isStrict ? ts.createNull() : ts.createIdentifier("_G")) : undefined;
+
+        // Argument evaluation.
+        const parameters = this.transformArguments([stringArray, ...expressions], signature, context);
+
+        // Assign raw strings to the "raw" property of the string table.
+        const tableParameterIndex = useSelfParameter ? 1 : 0;
+        const tableLiteralExpression =
+            parameters.length > tableParameterIndex &&
```

Comment:
Is there a case when it's not true?

Reply:
Yep, when there is no self parameter the table containing all the strings will be the first (0th) argument

Reply:
This unnecessary check has now been removed